packages/features/ee/users/components/UsersTable.tsx

```diff
@@ -1,20 +1,52 @@
-import { useState } from "react";
+import { useState, useRef, useMemo, useCallback, useEffect } from "react";
 
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { useDebounce } from "@calcom/lib/hooks/useDebounce";
 import { trpc } from "@calcom/trpc/react";
-import { Badge, ConfirmationDialogContent, Dialog, DropdownActions, showToast, Table } from "@calcom/ui";
-import { Edit, Trash } from "@calcom/ui/components/icon";
+import {
+  Badge,
+  ConfirmationDialogContent,
+  Dialog,
+  DropdownActions,
+  showToast,
+  Table,
+  TextField,
+  Avatar,
+} from "@calcom/ui";
+import { Edit, Trash, Lock } from "@calcom/ui/components/icon";
 
 import { withLicenseRequired } from "../../common/components/LicenseRequired";
 
-const { Body, Cell, ColumnTitle, Header, Row } = Table;
+const { Cell, ColumnTitle, Header, Row } = Table;
+
+const FETCH_LIMIT = 25;
 
 function UsersTableBare() {
+  const tableContainerRef = useRef<HTMLDivElement>(null);
   const utils = trpc.useContext();
-  const [data] = trpc.viewer.users.list.useSuspenseQuery();
+  const [searchTerm, setSearchTerm] = useState<string>("");
+  const debouncedSearchTerm = useDebounce(searchTerm, 500);
+
   const mutation = trpc.viewer.users.delete.useMutation({
     onSuccess: async () => {
       showToast("User has been deleted", "success");
-      await utils.viewer.users.list.invalidate();
+      // Lets not invalidated the whole cache, just remove the user from the cache.
+      // usefull cause in prod this will be fetching 100k+ users
+      utils.viewer.admin.listPaginated.setInfiniteData({ limit: FETCH_LIMIT }, (data) => {
+        if (!data) {
+          return {
+            pages: [],
+            pageParams: [],
+          };
+        }
+        return {
+          ...data,
+          pages: data.pages.map((page) => ({
+            ...page,
+            rows: page.rows.filter((row) => row.id !== userToDelete),
+          })),
+        };
```

Comment:
We manipulate the cache here to remove it instead of potentially fetching up to the current cursor. This could be like 50k users down the list. Not ideal.